modules/system/applications.nix

```diff
@@ -22,14 +22,26 @@ in
 
     system.activationScripts.applications.text = ''
       # Set up applications.
-      echo "setting up ~/Applications..." >&2
+      echo "setting up /Applications/Nix Apps..." >&2
+
+      ourLink () {
+        local link
+        link=$(readlink "$1")
+        [ -L "$1" ] && [ "${link#*-}" = 'system-applications/Applications' ]
+      }
+
+      # Clean up for links created at the old location in HOME
+      if ourLink ~/Applications; then
+        rm ~/Applications
```

Comment:
Only if it's a link left over from the previous setup, @ggPeti.

Reply:
That's what the `ourLink` function's supposed to check for.

Reply:
It's actually explained in the OP and the commit message:

> Because previous code will likely leave a symlink at
~/Applications/{,Nix Apps}, which will become dangling once the path
it links to is garbage collected from the store we test to see if a link
exists and it conforms to the path we're expecting and if it does remove
it.